clap_mangen/src/lib.rs

```diff
@@ -92,6 +92,51 @@ impl Man {
     }
 }
 
+/// Handle [`Man`] in relation to files
+impl Man {
+    /// Generate the filename of the manual page
+    #[must_use]
+    pub fn get_filename(&self) -> String {
+        format!(
+            "{}.{}",
+            self.cmd
+                .get_display_name()
+                .unwrap_or_else(|| self.cmd.get_name()),
+            self.section
+        )
+    }
+
+    /// [Renders](Man::render) the manual page and writes it to a file
+    pub fn generate_to<Dir>(&self, out_dir: Dir) -> Result<std::path::PathBuf, std::io::Error>
+    where
+        Dir: AsRef<std::path::Path>,
+    {
+        let filepath = out_dir.as_ref().join(self.get_filename());
+        let mut file = std::fs::File::create(&filepath)?;
+        self.render(&mut file)?;
+        file.flush()?;
+        Ok(filepath)
+    }
+}
+
+/// Generate manual page files for the command with all subcommands
+pub fn generate_to<Dir>(cmd: clap::Command, out_dir: Dir) -> Result<(), std::io::Error>
+where
+    Dir: AsRef<std::path::Path>,
+{
+    fn generate(cmd: clap::Command, out_dir: &std::path::Path) -> Result<(), std::io::Error> {
+        for cmd in cmd.get_subcommands().cloned() {
```

Comment:
Should we include or skip flattened subcommands?

Reply:
I tried my code on some of the example that use flattened subcommands and the output looks like something I would expect it to look like. The flatten logic seems to happen before `.get_subcommands()` so only subcommands shown in --help are actually passed here.

Reply:
> . The flatten logic seems to happen before .get_subcommands() so only subcommands shown in --help are actually passed here.

... that doesn't sound right.  Oh, are you referring to `#[command(flatten)]`?  I'm referring to [`Command::flatten_help`](https://docs.rs/clap/latest/clap/struct.Command.html#method.flatten_help).  See the `stash` command on the [git cookbook entry](https://docs.rs/clap/latest/clap/_derive/_cookbook/git_derive/index.html).

Whichever way we go, we should have a test for `flatten_help` to show that we went that way.

Reply:
To me `flatten_help` sounds unrelated to the file generation?

> To see more information, a user can still pass --help to the individual subcommands.

I would read this as "the information is still available at the individual subcommand". Therefore, I would still generate that subcommand man page?

Reply:
I suspect we'll decide to ignore `flatten_help` but I think we should be deliberate about it.  For example, if we look to `git` as a model, `git-stash` has a man page but not `git-stash-pop`.

For us `parent child --help` shows the full help while `parent --help` show's `child`s short help.  I wonder if we should have the man page show the full help for child commands.

In writing this out, I'm realizing this doesn't matter at all because there is no `flatten_help` support in `clap_mangen` so we have to output all commands so that everything is represented.  We can move this conversation to an issue which I'll create.